<!--StartFragment-->

## **HackBio Genomic Epidemiology Internship 2024**

Welcome!! This repository contains the tasks, skills and experience I accumulated during the course of the Genomic Epidemiology internship I completed with HackBio in 2024. The internship was designed to give hands-on experience and advance bioinformatics skills through personal or collaborative team projects using real-world data.  

![](https://lh7-rt.googleusercontent.com/docsz/AD_4nXeCkXc-LCprXfdXzBpf3ic02I5y99kiVBer48i80KlBqI_4Qegpyb2m0cTOExqZ-R3O9fGUOitMfBra0HNCgcJ9rErKSMqf9tgcxGa_Tra0M-NXGkeb1vKUwsQiijhlKTXbzr1NkT4s48NU6AKgF848P4x4?key=utQItuPyeST3ci2IFzNElvhe)

\


### **About HackBio**

HackBio is a community of experts across core data science competencies in life sciences, who combine industry and academic experiences to create up-to-date learning content needed to stay ahead in the bioinformatics industry. During the internship, interns get access to mentorship, collaboration, and hands-on learning experience, allowing interns to practice core skills through projects using real-world data.

### **Internship Track: AMR in Cancer (Genomic Epidemiology)**

Through the projects that were given during the internship, I used techniques such as R programming and Bash scripting to accomplish them. Through collaboration with other team members, I contributed to the improvement of the visualization and reporting of AMR and AMR products in the clinical trials pipeline from the World Health Organization. The internship was categorized into 5 stages (Stage 0 to 4).

#### **Stage 0**

In this task, I did a simple literature review on AMR in cancer epidemiology.

Link to task: <https://github.com/ChijiokeUhegwu/hackbio-cancer-internship/blob/main/AMR%20in%20cancer.md>

#### **Stage 1**

I collaborated in a team of 4 to analyze and draw insights from this study: [Whole-Genome Sequencing Accurately Identifies Resistance to Extended-Spectrum β-Lactams for Major Gram-Negative Bacterial Pathogens - PMC](https://pmc.ncbi.nlm.nih.gov/articles/PMC5850535/)

Link to task: [hackbio-cancer-internship/WGS(AMR\_in\_Cancer).md at main · MernaSalem/hackbio-cancer-internship](https://github.com/MernaSalem/hackbio-cancer-internship/blob/main/WGS\(AMR_in_Cancer\).md)

#### **Stage 2**

In this stage, I collaborated in a team of 5 to improve the visualization and reporting of AMR and AMR products in the clinical trials pipeline from the World Health Organization (WHO) using R programming, with the aim of creating intuitive and interactive analytics to describe the key insights from the dataset. 

Link to task: [hackbio-cancer-internship-/stage2\_task\_Report.md at main · MernaSalem/hackbio-cancer-internship-](https://github.com/MernaSalem/hackbio-cancer-internship-/blob/main/stage2_task_Report.md)

#### **Stage 3**

In this stage, I got Introduced to bioinformatics using Bash scripting for a simulated wet-lab biologist. I used bash scripting for bioinformatics tasks such as file navigation, sequencing data analysis, efficient identification of genetic sequences, file management, and manipulation of large biological datasets. Also, I used R programming to clean and  improve the visualization and reporting of cholera outbreak data from the World Health Organization (WHO), providing insightful summaries and reports for the global outbreak and a selected country (Nigeria).

Link to task: [https://github.com/ChijiokeUhegwu/hackbio-cancer-internship/tree/main/Stage%203.](https://github.com/ChijiokeUhegwu/hackbio-cancer-internship/tree/main/Stage%203)

#### **Stage 4**

In this stage, I developed a robust NGS pipeline using Bash scripting that can handle raw sequence reads from quality control to variant calling. The pipeline is as follows Quality Control (**FastQC**) => Trimming (**FastP**) => Genome Mapping (**bwa**) => Variant Calling (**bcftools**)

Link to task: <https://github.com/ChijiokeUhegwu/hackbio-cancer-internship/tree/main/Stage%204>

### **Conclusion**

The HackBio Genomic Epidemiology Internship 2024 provided me with a unique opportunity to develop bioinformatics skills by dealing with real-world data related to AMR in cancer and cholera outbreaks. I would say the internship was challenging and intense as I had to quickly pick up on the skills needed to deliver on the projects assigned to me. Through these projects, I gained hands-on experience using techniques and tools in R programming and Bash scripting to analyze complex datasets and build automated workflows. My GitHub and markdown skills also improved during the course of the internship. Shoutout to HackBio for the opportunity given to me to develop core skills needed in bioinformatics research, here is to more wins. 




<!--EndFragment-->
